### PR TITLE
AppArmor: support path globs when merging profiles

### PR DESCRIPTION
**Pull request by mhils**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR makes `spoc merge` understand glob patterns in the profile. This improves the UX when merging automatically recorded profiles into hand-maintained ones:

```shell
$ cat demo.yml
apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
kind: AppArmorProfile
spec:
  abstract:
    executable:
      allowedExecutables:
      - /sbin/ip{,6}tables-save
$ cat recorded.yml
apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
kind: AppArmorProfile
spec:
  abstract:
    executable:
      allowedExecutables:
      - /sbin/iptables-save
      - /sbin/ip6tables-save
$ build/spoc merge demo.yml recorded.yml
20:14:27.951272 Merging 2 profiles into /tmp/profile.yaml
20:14:27.951288 Reading file demo.yml
20:14:27.951582 Reading file recorded.yml
$ cat /tmp/profile.yaml
apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
kind: AppArmorProfile
metadata:
  creationTimestamp: null
spec:
  abstract:
    executable:
      allowedExecutables:
      - /sbin/ip{,6}tables-save. # no new entries
status: {}
```

#### Which issue(s) this PR fixes:

None

#### Does this PR have test?

Yes.

#### Does this PR introduce a user-facing change?

```release-note
`spoc merge` now combines AppArmor profiles with glob patterns in the first profile.
```

### CODE REVIEW

File: internal/pkg/manager/recordingmerger/apparmor.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
Comment (saschagrunert) [inline internal/pkg/manager/recordingmerger/apparmor.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

File: internal/pkg/manager/recordingmerger/apparmor_test.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
Comment (saschagrunert) [inline internal/pkg/manager/recordingmerger/apparmor_test.go:2]:
```suggestion
Copyright 2024 The Kubernetes Authors.
```

File: internal/pkg/manager/recordingmerger/apparmor.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+	}
+	return nil
+}
+
+func (m *appArmorPathSet) Add(pattern string) {
+	rex, err := appArmorGlobToRegex(pattern)
+	if err != nil {
+		log.Printf("Failed to parse AppArmor glob pattern '%s': %x\n", pattern, err)
+	}
+	m.paths = append(m.paths, apparmorPath{
```
Comment (ccojocar) [inline internal/pkg/manager/recordingmerger/apparmor.go:215]: nit: this is going to merge paths even though an error occurs, is this as expected?

**Review by ccojocar — APPROVED** [review]

**Comment by saschagrunert** [comment]
@mhils please rebase on top of `main` to get the latest CI fixes in.

File: internal/pkg/manager/recordingmerger/apparmor.go (hunk 3 above)
Comment (mhils) [reply internal/pkg/manager/recordingmerger/apparmor.go:215]: Yes. If `appArmorGlobToRegex` fails, we still want to add it to match on the raw pattern (just not on the regular expression). :)

**Review by saschagrunert — APPROVED** [review]

**Comment by mhils** [comment]
Looks like CI failure are unrelated to the PR and need a restart. :)

**Pull request merged**